src/commands/stack/StackForm.ts

```diff
@@ -0,0 +1,272 @@
+    if (!StackForm.instance) {
+      StackForm.instance = new StackForm();
+    }
+
+    return StackForm.instance;
+  }
+
+  constructor() {
+    super();
+
+    if (WebviewBase.context === null) {
+      throw new Error('Extension Context Not Propagated');
+    }
+
+    this.root = vscode.Uri.joinPath(WebviewBase.context.extensionUri, ...ROOT_PATH);
+    this.javaScript = vscode.Uri.joinPath(this.root, JS_FILE);
+    this.css = vscode.Uri.joinPath(this.root, CSS_FILE);
+
+    handlebars.registerHelper('capitalize', (str: string) => {
+      return str
+        .split('_')
+        .map(word => word[0].toUpperCase() + word.slice(1).toLowerCase())
+        .join(' ');
+    });
+
+    this.template = handlebars.compile(this.produceTemplate());
+  }
+
+  /**
+   * Opens a webview panel with a form to create a new stack
+   */
+  public async createForm() {
+    const panel = await this.display();
+    panel.webview.postMessage({ command: 'create' });
+  }
+
+  /**
+   * Opens a webview panel with a form to update a specified stack
+   * @param {string} id The id of the specified stack
+   * @param {string} name The current name of the specified stack
+   * @param {object} components The component settings of the sepcified stack
+   */
+  public async updateForm(id: string, name: string, components: { [type: string]: string }) {
+    const panel = await this.display();
+    panel.webview.postMessage({ command: 'update', data: { id, name, components } });
+  }
+
+  private async display(): Promise<vscode.WebviewPanel> {
+    const panels = Panels.getInstance();
+    const existingPanel = panels.getPanel('stack-form');
+    if (existingPanel) {
+      existingPanel.reveal();
+      return existingPanel;
+    }
+
+    const panel = panels.createPanel('stack-form', 'Stack Form', {
+      enableForms: true,
+      enableScripts: true,
+      retainContextWhenHidden: true,
+    });
+
+    await this.renderForm(panel);
+    this.attachListener(panel);
+    return panel;
+  }
+
+  private attachListener(panel: vscode.WebviewPanel) {
+    panel.webview.onDidReceiveMessage(
+      async (message: { command: string; data: { [key: string]: string } }) => {
+        let success = false;
+        const data = message.data;
+        const { name, id } = data;
+        delete data.name;
+        delete data.id;
+
+        switch (message.command) {
+          case 'create':
+            success = await this.createStack(name, data);
+            break;
+          case 'update':
+            const updateData = Object.fromEntries(
+              Object.entries(data).map(([type, id]) => [type, [id]])
+            );
+            success = await this.updateStack(id, name, updateData);
+            break;
+        }
+
+        if (!success) {
+          panel.webview.postMessage({ command: 'fail' });
+          return;
+        }
+
+        panel.dispose();
+        StackDataProvider.getInstance().refresh();
+      }
+    );
+  }
```

Comment:
The intent is to remove the properties from the `data` object. This is so that when they are sent to the Language Server they will not exist as the receiver is not expecting them.